src/DotVVM.Framework/Compilation/ViewCompilingVisitor.cs

```diff
@@ -39,21 +42,25 @@ public override void VisitView(ResolvedTreeRoot view)
 
             var createsCustomDerivedType = view.Metadata.Type == typeof(DotvvmView);
 
+            string resultControlType;
             if (createsCustomDerivedType)
             {
-                emitter.ResultControlType = className + "Control";
-                emitter.EmitControlClass(view.Metadata.Type, emitter.ResultControlType);
+                resultControlType = className + "Control";
+                emitter.EmitControlClass(view.Metadata.Type, resultControlType);
             }
             else
-                emitter.ResultControlType = view.Metadata.Type.FullName;
-
+            {
+                resultControlType = view.Metadata.Type.FullName;
```

Comment:
In this case, the type is always a full name. It would be simpler to call `emitter.ParseTypeName(view.Metadata.Type.)` directly instead of finding the type in the compilation again.